### PR TITLE
Add instance for test to top level API

### PR DESCRIPTION
**Pull request by dpeng817**

### CODE REVIEW

**Comment by dpeng817** [comment]
Current dependencies on/for this PR:
* master
  * **PR #10709** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/10709" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/10709?utm_source=stack-comment).

**Review by sryza — COMMENTED** [review]
I think it's worth adding a short docstring and adding it to apidocs - maybe https://docs.dagster.io/_apidocs/utilities?

File: python_modules/dagster/dagster/_core/test_utils.py [hunk 1]
```diff
@@ -119,7 +118,26 @@ def environ(env):
...
 @contextmanager
-def instance_for_test(overrides=None, set_dagster_home=True, temp_dir=None):
+def instance_for_test(
+    overrides: Optional[Mapping[str, Any]] = None,
+    set_dagster_home: bool = True,
+    temp_dir: Optional[str] = None,
+) -> Generator[DagsterInstance, None, None]:
+    """Creates a persistent :py:class:`~dagster.DagsterInstance` available within a context manager.
+
+    When a context manager is opened, if no `temp_dir` parameter is set, a new temporary directory will be created for the duration of the context manager's opening. If the `set_dagster_home` parameter is set to True (True by default), the `$DAGSTER_HOME` environment variable will be overridden to be this directory (or the directory passed in by `temp_dir`) for the duration of the context manager being open.
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/test_utils.py:128]: Line wrap plz. Same with below.

**Review by sryza — COMMENTED** [review]
To get into the API docs, this needs to be included in an rst file. Aside from that and the line wrap comment, this looks good to go.

**Review by sryza — APPROVED** [review]
Nice

**Pull request merged**